src/zenml/integrations/deepchecks/materializers/deepchecks_dataset_materializer.py

```diff
@@ -39,23 +39,23 @@ def handle_input(self, data_type: Type[Any]) -> Dataset:
         Returns:
             A Deepchecks Dataset.
         """
-        super().handle_input(data_type)
+        super().load(data_type)
 
         # Outsource to pandas
-        pandas_materializer = PandasMaterializer(self.artifact)
-        df = pandas_materializer.handle_input(data_type)
+        pandas_materializer = PandasMaterializer(self.uri)
```

Comment:
Wait what, why does this materializer not simply subclass the pandas one? :D

Reply:
good point, let me rewrite that

Reply:
Ah, the issue is that `save(x)` checks `if not self._can_handle_type(type(x)): ...`. If we subclass the pandas materializer and call `super().save(df.data)`, this would then assert that `df.data` is of type `Dataset` which it is not.

Reply:
This means materializer inheritance does not work as expected at the moment. Any suggestions how to handle this? Is it even worth fixing?

Personally, I feel like the proper solution would be to have `save()` and `load()` call two abstract functions `_save()` and `_load()` that perform just the data saving/loading without additional validations but that would be a major breaking change.

Reply:
I made the main changes on a separate branch, what do you guys think @htahir1 @schustmi @bcdurak?

https://github.com/zenml-io/zenml/pull/1175

Reply:
Another thing, I just realized the current PR would break custom materializers anyways since overwriting `handle_return()` and `handle_input()` will not do anything once our code calls `load()` and `save()`.

Reply:
For now we decided to keep the inheritance as it is. In order to support deprecated custom materializers, I added some logic that routes `save()` and `load()` to those legacy methods if they were overridden.